src/logging.cc

```diff
@@ -337,6 +338,16 @@ const char* GetLogSeverityName(LogSeverity severity) {
   return LogSeverityNames[severity];
 }
 
+LogSeverity FindFilepathLogSeverity(const std::string& filepath) {
+  for (int i = GLOG_INFO; i < NUM_SEVERITIES; i++) {
+    if (filepath.find(GetLogSeverityName(static_cast<LogSeverity>(i))) != std::string::npos) {
```

Comment:
```suggestion
    if (filepath.rfind(GetLogSeverityName(static_cast<LogSeverity>(i))) != std::string::npos) {
```
Search from right since the severity is used as a suffix.